fairlearn/preprocessing/_information_filter.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+from sklearn.preprocessing import StandardScaler
+
+
+def _scalar_projection(vec, unto):
+    return vec.dot(unto) / unto.dot(unto)
+
+
+def _vector_projection(vec, unto):
+    return _scalar_projection(vec, unto) * unto
+
+
+class InformationFilter(BaseEstimator, TransformerMixin):
+    r"""
+    The `InformationFilter` filters out sensitive correlations in a dataset.
+
+    The `InformationFilter` uses a variant of the Gram-Schmidt process
+    to filter information out of the dataset. This can be useful if you
+    want to filter information out of a dataset because of fairness.
+    To explain how it works: given a training matrix :math:`X` that contains
+    columns :math:`x_1, ..., x_k`. If we assume columns :math:`x_1` and :math:`x_2`
+    to be the sensitive columns then the information-filter will
+    remove information by applying these transformations;
+
+    .. math::
+
+       \\begin{split}
+       v_1 & = x_1 \\\\
+       v_2 & = x_2 - \\frac{x_2 v_1}{v_1 v_1}\\\\
+       v_3 & = x_3 - \\frac{x_k v_1}{v_1 v_1} - \\frac{x_2 v_2}{v_2 v_2}\\\\
+       ... \\\\
+       v_k & = x_k - \\frac{x_k v_1}{v_1 v_1} - \\frac{x_2 v_2}{v_2 v_2}
+       \\end{split}
+
+    Concatenating our vectors (but removing the sensitive ones) gives us
+    a new training matrix :math:`X_{filtered} =  [v_3, ..., v_k]`.
+
+    :param columns: list of columns to filter out this can be a sequence of either int
+                    (in the case of numpy) or string (in the case of pandas).
+    :param alpha: parameter to control how much to filter, for alpha=1 we filter out
```

Comment:
How does `\alpha` affect the equations above?

Reply:
It smoothes between the original columns and the newly created ones. I could add some maths if that makes it clearer.

Reply:
It might be more of a userguide thing, but I think it would be helpful to have the way it enters the equations shown.